distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1310,12 +1405,67 @@ private String ingestXmlWithProduct(String fileName) throws IOException {
         return metacardId;
     }
 
-    private void cleanProductCache() throws IOException {
-        FileUtils.cleanDirectory(Paths.get(ddfHome).resolve(PRODUCT_CACHE).toFile());
+
+    private Action getCswRetrievalHeaders(String filename) {
+        return composite(header("X-Csw-Product", "true"), header("Content-Disposition", "filename=" + filename));
+    }
+
+    private void verifyJson(JsonPath json, String attributePath, String expectedValue) {
+        String value = json.getString(attributePath);
+        assertThat(value, is(expectedValue));
+    }
+
+    private void verifyActivity(JsonPath json, String title, String message, String status) {
+        verifyJson(json, "data.title", title);
+        verifyJson(json, "data.message", message);
+        verifyJson(json, "data.status", status);
+    }
+
+    private void verifyNotification(JsonPath json, String title, String message, String status) {
+        verifyJson(json, "data.title", title);
+        verifyJson(json, "data.message", message);
+        verifyJson(json, "data.status", status);
+    }
+
+    private List<String> sortJsonMessagesByTimestamp(List<String> unsortedMessageList) {
+        Map<LocalDateTime, String> unsortedMessageMap = new HashMap<>();
```

Comment:
Couldn't we use `Collections.sort(List, Comparator)` instead and have the `Comparator` extract and use the timestamp to order the messages? We could also add a `getSortedMessages()` on the `CometDClient` class instead of doing this here?